docs/WITH_TYPESCRIPT.md

```diff
@@ -0,0 +1,9 @@
+# Example to work with TypeScript
+
+_This is non-normative guide to use this presets with TypeScript.
+By changing versions of each of plugins, this guide might not works_.
+
+
+## `eslint-plugin-import`
+
+Add the following config for your 
```

Comment:
I seem it's better to write the following sample code. But I have not much time to figured out yet the details of why this code works and why we don't have to use....

```javascript
'use strict';

const JS_EXTENSION = ['.js', '.jsx'];
const TS_EXTENSION = ['.ts', '.tsx'];
const ALL_EXTENSION = [...JS_EXTENSION, ...TS_EXTENSION];

module.exports = {
    'settings': {
        // It might be better to _not_ specifying this if you use TypeScript with some type checker.
        // Then you can detect what a symbol is exported from a module by static checking by the typechecker.
        'import/extensions': ALL_EXTENSION,

        // I don't seem that you don't have to use this
        // if you use `'@typescript-eslint` plugin to lint your typescript.
        // Then eslint-plugin-impor will not switch the parser and use the more globally specified.
        'import/parsers': {
            '@typescript-eslint/parser': TS_EXTENSION,
        },

        'import/resolver': {
            'node': {
                'extensions': ALL_EXTENSION,
            },
        }
    },
};
```

Reply:
IMO, I seem it's safe to only specify `import/resolver` for TypeScript project....